packages/subgraph/src/v1-analytics/mapping.ts

```diff
@@ -291,8 +332,52 @@ export function handleTransactionCancelled(event: TransactionCancelled): void {
       assetBalance.assetId = transaction.receivingAssetId;
       assetBalance.router = event.params.router.toHex();
       assetBalance.amount = new BigInt(0);
+      assetBalance.supplied = new BigInt(0);
+      assetBalance.locked = new BigInt(0);
     }
     assetBalance.amount = assetBalance.amount.plus(transaction.amount);
+    assetBalance.locked = assetBalance.locked.sub(transaction.amount);
     assetBalance.save();
   }
+
+  // update metrics
+  let timestamp = event.block.timestamp.toI32();
+
+  let hour = timestamp / 3600; // rounded
+  let hourStartTimestamp = hour * 3600;
+
+  let hourIDPerAsset = hour.toString() + "-" + transaction.receivingAssetId.toHex();
+
+  let hourlyMetric = HourlyMetric.load(hourIDPerAsset.toString());
+  if (hourlyMetric === null) {
+    hourlyMetric = new HourlyMetric(hourIDPerAsset.toString());
```

Comment:
let hourIDPerAsset = hour.toString().concat("-").concat(transaction.receivingAssetId.toHex());

  let hourlyMetric = HourlyMetric.load(hourIDPerAsset);
  if (hourlyMetric === null) {
    hourlyMetric = new HourlyMetric(hourIDPerAsset);
    
    I think it could be like this as well. hourIDPerAsset is string object right now. any reason to use `toString` here?